apps/labs/posts/scikit-lego-narwhals.md

```diff
@@ -0,0 +1,101 @@
+---
+title: 'How Narwhals and scikit-lego came together to enable dataframe-agnosticism'
+authors: [marco-gorelli]
+published: May 26, 2024
+description: 'And how your library can become dataframe-agnostic too'
+category: [PyData ecosystem]
+---
+
+# How scikit-lego became dataframe-agnostic
+
+Scikit-lego is a Python project which provides extra building blocks on top of scikit-learn.
+It provides some extra functionality such as `TimeGapSplit` and `add_lags` which may be considered
```

Comment:
Might be nice to show the big guns and some stats here? I think our grouped estimators are probably the most popular features, and it could also be good to emphesize that it's not a small project. 

#### So maybe something like:

Scikit-lego is a Python project from the scikit-learn ecosystem that contributes extra estimators for machine-learning pipelines. It's a relatively popular project with over 1000 Github stars and 20,000 downloads a month. It's popularity comes from industry as many of the tools that it provides aren't proven to be state-of-the-art, but fall a bit more in the 'tricks that work' category. These features include a [GroupedEstimator](https://koaning.github.io/scikit-lego/user-guide/meta-models/#grouped-prediction) which can effectively run full pipelines split per subset, a [TimeGapSplit](https://koaning.github.io/scikit-lego/user-guide/cross-validation/#timegapsplit) cross validator for timeseries that adds a gap between train and test set and a whole suite of techniques that leverage [mixture methods](https://koaning.github.io/scikit-lego/user-guide/mixture-methods/) for outlier detection and non-linear classification.

Recently, because polars has been gaining traction, the library maintainers have been looking for a simple way to support multiple dataframe implementations. The project still wants to support pandas, but it would be a shame if newer dataframes couldn't be supported.

Enter narwhals. As of scikit-lego 0.9.0 ...